modules/afmongodb/afmongodb-worker.c

```diff
@@ -353,6 +402,30 @@ _vp_process_value(const gchar *name, const gchar *prefix, LogMessageValueType ty
   return FALSE;
 }
 
+static LogThreadedResult
+_worker_batch_flush(LogThreadedDestWorker *s, LogThreadedFlushMode expedite)
+{
+  MongoDBDestWorker *self = (MongoDBDestWorker *) s;
+
+  bson_error_t error;
+  int result = mongoc_bulk_operation_execute(self->bulk_op, NULL, &error);
```

Comment:
`reply` is not optional:

https://mongoc.org/libmongoc/current/mongoc_bulk_operation_execute.html

Reply:
actually it is ;)
the documentation says nothing about optionallity, it say, is always initialized, even upon failure
if this parameter is not presented than it will not be used, allocated, there's nothing to free via bson_destroy()
it is working nicely this way, i did not see a crash or leak at all

Reply:
double checked meanwhile in the mongo-c-driver source
reply safely can be NULL, it is always guarded by `_mongoc_bson_init_if_set (reply);`

Reply:
2 rows below:
```
error: An *optional* location for a [bson_error_t] or NULL.
```

What is not documented is subject to change. I would go the safe route.

Reply:
yeep, I saw that too, but that documentation is outdated, and full of wrong information, that is why I started to check the source code itself for everything I had doubts about
as you wish, but it is a unnecessary overhead here both in speed and allocations